packages/fluentui/react-northstar-prototypes/src/prototypes/instructionMessage/instructionMessage.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useFluentContext } from '@fluentui/react-bindings';
+import { Box, Ref } from '@fluentui/react-northstar';
+
+interface InstructionMessageProps {
+  message: string;
+  role?: string;
+  'aria-label'?: string;
+}
+
+const InstructionMessage: React.FunctionComponent<InstructionMessageProps> = props => {
+  const { children, message, ...rest } = props;
+
+  const narrationHappened = React.useRef(false);
+  const instructionRef = React.useRef<HTMLElement>();
+  const timeoutMessageNarrate = React.useRef<number>();
+  const timeoutMessageRemove = React.useRef<number>();
+  const context = useFluentContext();
+  const areaFocusableElements = () =>
+    instructionRef?.current?.querySelectorAll('a, button, input, textarea, select, details, [tabindex]');
+
+  const narrate = (instructionMessage, priority = 'polite') => {
+    const element = document.createElement('div');
+    element.setAttribute(
+      'style',
+      'position: absolute; left: -10000px; top: auto; width: 1px; height: 1px; overflow: hidden;',
+    );
+    element.setAttribute('aria-live', priority);
+    document.body.appendChild(element);
+
+    timeoutMessageNarrate.current = context.target.defaultView.setTimeout(() => {
+      element.innerText = instructionMessage;
+    }, 2000);
+
+    timeoutMessageRemove.current = context.target.defaultView.setTimeout(() => {
+      document.body.removeChild(element);
+    }, 2300);
+  };
+
+  const handleFocus = React.useCallback(
+    event => {
+      const focElements = areaFocusableElements();
+      if (!narrationHappened.current && focElements && Array.from(focElements).includes(event.target)) {
+        narrate(message);
+        narrationHappened.current = true;
+      }
+    },
+    [areaFocusableElements],
+  );
+
+  const handleBlur = React.useCallback(
+    event => {
+      const focElements = areaFocusableElements();
+      if (areaFocusableElements && !Array.from(focElements).includes(event.relatedTarget)) {
```

Comment:
```suggestion
      if (focElements && !Array.from(focElements).includes(event.relatedTarget)) {
```
I think this is a typo?

Reply:
I'm kinda confused by the 2 `if` in handleFocus/handleBlur.

This is how I understand handleFocus:
when narration hasn't been done + there's focusable DOM in the area + one of those DOM is focused -> then we narrate instruction msg

For handleBlur, I don't understand why condition here is
`&& !Array.from(focElements).includes(event.relatedTarget)` 
For me, it means, the blur event didn't happen within the DOM in the area, but happened somewhere else.

Reply:
Why "didn't happen within the DOM"? `event.relatedTarget` is the element that will/have received focus in the `blur`

Reply:
Oh now that make sense. Thanks Charles for explain